src/OpenApi/test/OpenApiGeneratorTests.cs

```diff
@@ -363,7 +363,7 @@ static void AssertParameters(OpenApiOperation operation)
                 operation.Parameters,
                 param =>
                 {
-                    Assert.Equal("Foo", param.Name);
+                    Assert.Equal("Foo", param.Name, ignoreCase: true);
```

Comment:
Maybe we don't need the line **394** or you can move it to your new set of tests to check the correct parameter name. Something like this:

```
    [Theory]
    [InlineData("/todos/{foo}", "foo")]
    [InlineData("/todos/{Foo}", "Foo")]
    [InlineData("/todos/{foo:minlen(2)}", "foo")]
    public void FavorsParameterCasingInRoutePattern_ForArgumentList(string pattern, string expectedName)
    {
        var operation = GetOpenApiOperation(([AsParameters] ArgumentListRecordWithoutAttributes req) => "", pattern);

        var param = Assert.Single(operation.Parameters);
        Assert.Equal(expectedName, param.Name);
    }
```

Reply:
Added test cases for this.